build/dsc/templates/dsc_type.rb.erb

```diff
@@ -86,11 +86,15 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
 
   ensurable do
     newvalue(:exists?) { provider.exists? }
-    newvalue(:<%= begin resource.ensure_value rescue 'present' end %>) { provider.create }
+<%  resource.valid_ensure_present_values.each do |property| -%>
+    newvalue(:<%= property.downcase %>) { provider.create }
+<%  end -%>
 <%  if resource.absentable? -%>
-    newvalue(:<%= resource.absent_value %>)  { provider.destroy }
+<%  resource.valid_ensure_absent_values.each do |property| -%>
+    newvalue(:<%= property.downcase %>)  { provider.destroy }
```

Comment:
Hopefully we never see multiple `absent` states... as that doesn't really make any sense.

Different creation states can make sense sometimes, but I'm not so sure the `absent` state parallel holds true.  

Thoughts on whether we should even attempt / address this?